tests/System.Text.Primitives.Tests/EncodingPerfTestsCoreCLR.cs

```diff
@@ -0,0 +1,146 @@
+    {
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [MemberData(nameof(GetEncodingPerformanceTestData))]
+        public void EncodeFromUtf8toUtf8UsingEncoding(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf8 = Encoding.UTF8;
+            int utf8Length = utf8.GetByteCount(characters);
+            var utf8Buffer = new byte[utf8Length];
+            utf8.GetBytes(characters, 0, characters.Length, utf8Buffer, 0);
+
+            var temp = utf8.GetChars(utf8Buffer);
+            byte[] output = Encoding.Convert(utf8, utf8, utf8Buffer);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                        utf8.GetBytes(utf8.GetChars(utf8Buffer), 0, temp.Length, output, 0);
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [MemberData(nameof(GetEncodingPerformanceTestData))]
+        public void EncodeFromUtf16toUtf8UsingEncoding(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf8 = Encoding.UTF8;
+            int utf8Length = utf8.GetByteCount(characters);
+            var utf8Buffer = new byte[utf8Length];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                        utf8.GetBytes(characters, 0, characters.Length, utf8Buffer, 0);
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [MemberData(nameof(GetEncodingPerformanceTestData))]
+        public void EncodeFromUtf32toUtf8UsingEncoding(int length, int minCodePoint, int maxCodePoint, SpecialTestCases special = SpecialTestCases.None)
+        {
+            string inputString = GenerateStringData(length, minCodePoint, maxCodePoint, special);
+            char[] characters = inputString.AsSpan().ToArray();
+            Encoding utf32 = Encoding.UTF32;
+            Encoding utf8 = Encoding.UTF8;
+            int utf32Length = utf32.GetByteCount(characters);
+            var utf32Buffer = new byte[utf32Length];
+            utf32.GetBytes(characters, 0, characters.Length, utf32Buffer, 0);
+
+            int utf8Length = utf8.GetByteCount(characters);
+
+            var temp = utf32.GetChars(utf32Buffer);
+            byte[] output = Encoding.Convert(utf32, utf8, utf32Buffer);
```

Comment:
Are you doing this as a way to pre-allocate memory for the calls below? If that is true, you have the necessary value to allocate directly via the utf8Length above, which is otherwise never used.